src/libraries/System.Private.CoreLib/src/System/Guid.cs

```diff
@@ -884,6 +932,81 @@ public bool TryWriteBytes(Span<byte> destination)
             return true;
         }
 
+        // Returns whether bytes are successfully written to given span.
+        public bool TryWriteBytes(Span<byte> destination, out int bytesWritten)
+        {
+            if (BitConverter.IsLittleEndian)
+            {
+                bool result = MemoryMarshal.TryWrite(destination, ref Unsafe.AsRef(in this));
+                bytesWritten = result ? 16 : 0;
+                return result;
+            }
+
+            // slower path for BigEndian
+            if (destination.Length < 16)
+            {
+                bytesWritten = 0;
+                return false;
+            }
+
+            destination[15] = _k; // hoist bounds checks
+            BinaryPrimitives.WriteInt32LittleEndian(destination, _a);
+            BinaryPrimitives.WriteInt16LittleEndian(destination.Slice(4), _b);
+            BinaryPrimitives.WriteInt16LittleEndian(destination.Slice(6), _c);
+            destination[8] = _d;
+            destination[9] = _e;
+            destination[10] = _f;
+            destination[11] = _g;
+            destination[12] = _h;
+            destination[13] = _i;
+            destination[14] = _j;
+
+            bytesWritten = 16;
+            return true;
+        }
+
+        // Returns whether bytes are successfully written to given span.
+        public bool TryWriteBytes(Span<byte> destination, bool bigEndian, out int bytesWritten)
```

Comment:
Do we need three different implementations of TryWriteBytes just to support the optional bool and out int, or could we consolidate to just two or one implementations?

Reply:
I think that's enough 2 implementations of `TryWriteBytes` but the approved API has such comment `// public bool TryWriteBytes(Span<byte> destination, out int bytesWritten); -- new in .NET 8`

I will remove the redundant method.

Reply:
I'm not suggesting removing a method, just having one delegate to the other or some such thing

Reply:
I added both methods.

`bool TryWriteBytes(Span<byte> destination, out int bytesWritten); -- new in .NET 8` is added but commented in the approved issue https://github.com/dotnet/runtime/issues/86798#issuecomment-1603081570

I think it is my mistake and only `public bool TryWriteBytes(Span<byte> destination, bool bigEndian, out int bytesWritten)` should be added.

Reply:
@bartonjs, @tannergooding, can you clarify what surface area is intended / approved?

Reply:
`TryWriteBytes(Span<byte>, out int)` was approved in #30940.

I didn't ask about "removing" it during the API Review for `TryWriteBytes(Span<byte>, bool, out int)` because I assumed it was already written (and thought it was more than 2 months ago that we approved it).

Personally I'm cool with omitting it and just forcing the caller who wants the out-int to specify the endianness bool.  But, if that's the case, we should close #30940.  (And if we implement it here, we should also close the issue)